File: 219-contains-duplicate-ii/contains-duplicate-ii.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        check = set()
        for i in range(len(nums)):
            if nums[i] in check:
                return True
            check.add(nums[i])
            
            if len(check) > k:
                check.remove(nums[i-k])
        return False

File: 219-contains-duplicate-ii/contains-duplicate-ii.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
var containsNearbyDuplicate = function(nums, k) {
let check = new Set()

    for(let i=0;i< nums.length;i++){
       if(check.has(nums[i])) {
        return true
       }
       check.add(nums[i])
       if(check.size > k){
        check.delete(nums[i-k])
       }
    }
     return false;
};